### ISSUE TITLE
some questions

### ISSUE THREAD

**Issue by xuefeng-xu** [Jaskiee/GAN-Attack-against-Federated-Deep-Learning#3]
Hi @Jaskiee, I was also curious about this "GAN attack" paper in some places.

Let's assume that the clients are training a classification model on the MNIST dataset. 
So the last layer of the model is a softmax layer with dimention=10.
Recall that the author said attacker use this model as the discriminator, 
but he didn't tell any detail about how to define the loss functions or 
how to represent p(y=fake | x).

I've read your code about the discriminator, and you set the last layer's dimention=11.

So here are my questions:
- Does this layer setting refers to the paper "Improved Techniques for Traning GANs"?
- Why do the clients train a model with the last layer's dimention=11?
- Have you tested your method under differential privacy?

By the way, I've sent an email to the author. Unfortunately, I didn't get a response yet.

**Comment by Jaskiee** [comment]
Thanks for your questions, I haven't followed up this topic for months so please let me take some time to review and read the paper you mentioned.  I will reply as soon as possible, hope that can help you!

**Comment by Jaskiee** [comment]
> Hi @Jaskiee, I was also curious about this "GAN attack" paper in some places.
> 
> Let's assume that the clients are training a classification model on the MNIST dataset.
> So the last layer of the model is a softmax layer with dimention=10.
> Recall that the author said attacker use this model as the discriminator,
> but he didn't tell any detail about how to define the loss functions or
> how to represent p(y=fake | x).
> 
> I've read your code about the discriminator, and you set the last layer's dimention=11.
> 
> So here are my questions:
> 
> * Does this layer setting refers to the paper "Improved Techniques for Traning GANs"?
> * Why do the clients train a model with the last layer's dimention=11?
> * Have you tested your method under differential privacy?
> 
> By the way, I've sent an email to the author. Unfortunately, I didn't get a response yet.

Thanks for your questions, I haven't followed up this topic for months so please let me take some time to review and read the paper you mentioned. I will reply as soon as possible, hope that can help you!

**Comment by jacobwin-ai** [comment]
I am wondering if this issue is sovled?
I read your code and failed to find any lines related to differencial privacy deployment.
Sincerely waiting for your response.

**Comment by xuefeng-xu** [comment]
> I am wondering if this issue is sovled?
> I read your code and failed to find any lines related to differencial privacy deployment.
> Sincerely waiting for your response.

Not yet. Still waiting.

**Comment by woohan** [comment]
Hi all, is this issue solved?

**Comment by jacobwin-ai** [comment]
您好，您的邮件我已收到，我将会及时阅读给您回复，谢谢！

**Comment by xuefeng-xu** [comment]
> Hi all, is this issue solved?

Nope. But I think the setting of the loss function was refer to the paper I metioned, which could answer my second question. For the last question, I think differential privacy couldn't defence this so-called GAN attack unless the injected noise is very large.

**Comment by xuefeng-xu** [comment]
In my opinion, this GAN attack is probmatic, cf. the oral presetation on youtube and the blog post mentioned by the questioner in the pre.
